tool_result_selection/models.py

```diff
@@ -55,126 +52,80 @@ def pop_up_unkown_datasource_type():
         "1. simulate this model again and load the result in QGIS3 \n"
         "2. load this result into QGIS2.18 ThreeDiToolbox v1.6 "
     )
-    # we only continue if self.datasource_type == 'netcdf-groundwater'
     logger.error(msg)
     pop_up_info(msg, title="Error")
-    raise AssertionError("unknown datasource type")
 
 
 class ValueWithChangeSignal(object):
-    def __init__(self, signal_name, signal_setting_name, init_value=None):
+    """Value for use inside a BaseModel. A change emits a signal.
+
+    It works like a python property. The whole ``__get__``, ``instance``,
+    ``owner`` stuff is explained here:
+    https://stackoverflow.com/a/18038707/27401
+
+    The ``signal_setting_name`` has to do with the way project state is saved,
+    see ``utils/qprojects.py``.
+
+    """
+
+    def __init__(self, signal_name, signal_setting_name, initial_value=None):
+        """Initialize ourselves as a kind-of-python-property.
+
+        ``signal_name`` is the name of a class attribute that should be a qtsignal.
+
+        ``signal_setting_name`` is the string that gets emitted as the first
+        argument of the signal. It functions as a key for the key/value state
+        storage mechanism from ``utils.qprojects.py``.
+
+        """
         self.signal_name = signal_name
         self.signal_setting_name = signal_setting_name
-        self.value = init_value
+        self.value = initial_value
 
-    def __get__(self, instance, type):
+    def __get__(self, instance, owner):
         return self.value
 
     def __set__(self, instance, value):
         self.value = value
         getattr(instance, self.signal_name).emit(self.signal_setting_name, value)
 
 
-class DataSourceLayerManager(object):
-    """
-    Abstracts away datasource-layer specifics.
-    """
-
-    type_datasource_mapping = {"netcdf-groundwater": ThreediResult}
-
-    def __init__(self, datasource_type, file_path):
-        self.datasource_type = datasource_type
-        self.file_path = file_path
+class DatasourceLayerHelper(object):
+    def __init__(self, file_path):
+        self.file_path = Path(file_path)
+        self.datasource_dir = self.file_path.parent
+        self.gridadmin_filepath = str(self.datasource_dir / "gridadmin.sqlite")
```

Comment:
`gridadmin_filepath` can be confusing with the `gridadmin.h5` file. 
This `gridadmin.h5` file is used by threedigrid.
`gridadmin.sqlite` is there because of historic reasons (threedigrid was not yet developed). They should serve the same purpose but I'm not sure if `gridadmin.sqlite` got more functionality over time.

It's quite unfortunate that these two files have the same name, and only differ in extension...

Maybe name it `sqlite_gridadmin_filepath` to highlight that this is the sqlite file and the old way?

Reply:
Goed idee. Gedaan.